client/src/pages/Room.jsx

```diff
@@ -1,25 +1,44 @@
 import { useParams } from "react-router-dom";
-import { useEffect, useState } from "react";
+import useWebSocket, { ReadyState } from "react-use-websocket";
+import { useEffect, useState, useMemo } from "react";
+import useUser from "../api/users";
 import useStyles from "./styles";
 
 function Room() {
-  const [users, setUsers] = useState([]);
-  const { name } = useParams();
+  const { name: room } = useParams();
+  const { user: myUser } = useUser();
+  const [{ users, creator }, setRoom] = useState({});
   const classes = useStyles();
 
-  useEffect(
-    () =>
-      fetch(`/room/get?${new URLSearchParams({ name })}`)
-        .then((res) => res.json())
-        .then((res) => setUsers(res.users)),
-    [name]
+  const { sendMessage, lastJsonMessage: roomObj, readyState } = useWebSocket(
+    `ws://localhost:4000/?${new URLSearchParams({ room })}`,
+    {
+      onOpen: () => console.log(`WebSocket connection to room ${room} opened`),
+      shouldReconnect: () => true,
+    }
   );
 
+  useMemo(() => roomObj && setRoom(roomObj), [roomObj]);
```

Comment:
I know I wrote this code, not you, but wanted to confirm that this is an optimization and not just useless code